tests/bluetooth/tester/src/audio/btp_cap.c

```diff
@@ -561,12 +555,15 @@ static int cap_broadcast_source_adv_setup(struct btp_bap_broadcast_local_source
 	base_ad[1].type = BT_DATA_NAME_COMPLETE;
 	base_ad[1].data_len = sizeof(CONFIG_BT_DEVICE_NAME) - 1;
 	base_ad[1].data = CONFIG_BT_DEVICE_NAME;
-	err = tester_gap_create_adv_instance(&param, BTP_GAP_ADDR_TYPE_IDENTITY, base_ad, 2, NULL,
-					     0, gap_settings);
-	if (err != 0) {
-		LOG_DBG("Failed to create extended advertising instance: %d", err);
 
-		return -EINVAL;
+	if (!cap_prohibit_setting_up_ext_ad) {
+		err = tester_gap_create_adv_instance(&param, BTP_GAP_ADDR_TYPE_IDENTITY, base_ad,
+						     2, NULL, 0, gap_settings);
+		if (err != 0) {
+			LOG_DBG("Failed to create extended advertising instance: %d", err);
+
+			return -EINVAL;
+		}
```

Comment:
Hmm, another thing to consider here is that `tester_gap_padv_configure` will always fail if `cap_broadcast_source_adv_setup` is called when `cap_prohibit_setting_up_ext_ad = false` and if no-one else has made the extended advertising. 

Makes me wonder if `if (!cap_prohibit_setting_up_ext_ad) {` should just have been replaced with `if (tester_gap_ext_adv_get == NULL)`

Reply:
That might work, was thinking a little about that yesterday.

Reply:
Trying it out now